web/src/client/network/network_manager.js

```diff
@@ -500,19 +503,40 @@ class NetworkManagerAdapter {
     return { address: data.address.v, prefix: parseInt(data.prefix.v) };
   }
 
-  settingsFromProxies(manager, settings) {
-    return {
-      wireless: !!(manager?.WirelessEnabled && manager?.WirelessHardwareEnabled),
-      hostname: settings?.Hostname || ""
-    };
+  /*
+  * Returns the list of WiFi devices available in the system
+  *
+  * @return {object[]} list of available WiFi devices
+  */
+  availableWifiDevices() {
+    return Object.values(this.proxies.devices).filter(d => d.DeviceType === NM_DEVICE_TYPE_WIFI);
+  }
+
+  /*
+  * Returns whether the system is able to scan wifi networks based on rfkill and the presence of
+  * some wifi device
+  *
+  * @return {boolean}
+  */
+  wifiScanSupported() {
+    const { manager } = this.proxies;
+
+    if (!manager) return false;
+    if (!(manager.WirelessEnabled && manager.WirelessHardwareEnabled)) return false;
+
+    return this.availableWifiDevices().length > 0;
   }
 
   /*
   * Returns NetworkManager general settings
+  *
```

Comment:
NP: Ah, I see we have a mix on comments style :) So keep them as they are :man_shrugging:

Reply:
I do not think these are mixed styles, just miss-aligned comments :) And sure we have more of them. 😞